### ISSUE TITLE
go1.23 support

### ISSUE THREAD

**Issue by ldez** [golangci/golangci-lint#4837]
**EDIT: since v1.60.0 golangci supports go1.23 :tada:**

---

`golangci-lint` is a free and open-source project built by volunteers.

If you value it, consider supporting us, we appreciate it! :heart:

[![Open Collective backers and sponsors](https://img.shields.io/badge/OpenCollective-Donate-blue?logo=opencollective&style=for-the-badge)](https://opencollective.com/golangci-lint) [![GitHub Sponsors](https://img.shields.io/badge/GitHub-Donate-blue?logo=github&style=for-the-badge)](https://github.com/sponsors/golangci)

https://github.com/golangci/golangci-lint/assets/5674651/95bcfd16-cc12-413d-a964-91f6d47ded87

_Video created by a human ([ldez](https://github.com/ldez)) with [Inkscape](https://inkscape.org), [Synfig](https://www.synfig.org/), [Kdenlive](https://kdenlive.org)._

---

#### Is golangci-lint support go1.23?

No, and yes.

Basically, golangci-lint supports Go versions lower or equal to the Go version used **to build it**.

| Go version (build) | Supported Go versions (code)     | Not supported Go versions (code) |
|--------------------|----------------------------------|----------------------------------|
| go1.21             | <= go1.21 (go1.21, go1.20, etc.) | > go1.21 (go1.22, go1.23, etc.)  |
| go1.20             | <= go1.20 (go1.20, go1.19, etc.) | > go1.20 (go1.21, go1.22, etc.)  |

The current binaries from the releases page and the Docker images have been built with go1.22 and will be built with go1.22 as long as go1.23 is not GA.
So those binaries/images don't support go1.23 (new elements and internal Go changes).

https://golangci-lint.run/usage/faq/#which-go-versions-are-supported

In addition, of the Go version used to build it, some linters could need to be updated to support newer versions of Go, and internal pieces of golangci-lint could also need to be updated.

You can follow the PR https://github.com/golangci/golangci-lint/pull/4836 to see the progress on this topic.

---

- [x] staticcheck, gosimple, stylecheck: waiting for a release. [details](https://github.com/dominikh/go-tools/issues/1523) 

---

[![Open Collective backers and sponsors](https://img.shields.io/badge/OpenCollective-Donate-blue?logo=opencollective&style=for-the-badge)](https://opencollective.com/golangci-lint) [![GitHub Sponsors](https://img.shields.io/badge/GitHub-Donate-blue?logo=github&style=for-the-badge)](https://github.com/sponsors/golangci)

**Comment by Zxilly** [comment]
Is that possible to push the pending commits for go 1.23 to other branch than master, so we can try the new version in the CI process？

**Comment by Zxilly** [comment]
Let me explain my idea in a little more detail:

We now have a PR https://github.com/golangci/golangci-lint/pull/4836 pointing to `golangci:master` from `ldez:feat/go1.23`. Can we change this PR to `golangci:feat/go1.23` pointing to `golangci:master`?

This will allow the user to install the version to be tested directly using `go install github.com/golangci/golangci-lint@feat/go1.23`, so the `goinstall` mode of [lint-action](https://github.com/golangci/golangci-lint-action) does not need to be changed to accommodate this change.

**Comment by ldez** [comment]
> We now have a PR https://github.com/golangci/golangci-lint/pull/4836 pointing to `golangci:master` from `ldez:feat/go1.23`. Can we change the base of this PR to `golangci:feat/go1.23`?

`ldez:feat/go1.23` is not the base branch but the "from" branch.

**Changing the "from" branch of an already open PR is impossible.**

The PR is a draft and will be rebased and modified several times during the RC phase, I recommend not using it in production.

I could close the PR and re-create one, but, if I do that, I also want to clarify that should not be considered as an official release.

**Comment by ldez** [comment]
Due to a problem (the Docker image `docker.io/library/golang:1.23-alpine` is not yet available, the release v1.60.0 has failed.

I will relaunch the release when the Docker image is available.

**Comment by DmitriyMV** [comment]
@ldez looks like 1.23-alpine is uploaded.

**Comment by ldez** [comment]
only amd64 is available.

We need arm64 too.

**Comment by ldez** [comment]
Still waiting for the manifests:
- https://doi-janky.infosiftr.net/job/wip/job/new/job/deploy-amd64/7461/changes
- https://doi-janky.infosiftr.net/job/wip/job/new/job/deploy-arm64v8/7624/changes

**Comment by ldez** [comment]
I'm cursed... now there is no space left on the CI... :cry:

**Comment by ldez** [comment]
I used an action to remove useless elements inside the CI:

```
BEFORE CLEAN-UP:

Filesystem      Size  Used Avail Use% Mounted on
/dev/root        73G   53G   21G  73% /

AFTER CLEAN-UP:

Filesystem      Size  Used Avail Use% Mounted on
/dev/root        73G   28G   45G  39% /
```

So we gain 24Go, which should fix the problem with the space during the release.
But I'm forced to create a v1.60.1.
The v1.60.0 is stuck :cry:.

It's 3:30 am for me, I've been on this release since 7:00 pm...
I will stay here until this release is done (and after sending all the announcements on social networks).

If you want to share my journey, I think I will launch a quick stream on Twitch https://www.twitch.tv/ldez_oss.

**Comment by ldez** [comment]
It's done :tada: 

I hope you will enjoy this release.

It's 4:00 am so I will go to sleep :zzz:

**Comment by ozancaglayan** [comment]
Hi,

this is still an issue for me. Using go 1.23, and a golangci-lint 1.60.3 built with 1.23, memory usage peaks at available RAM and the process gets killed. I don't understand what's the current requirement to fix this.

Thanks

**Comment by nesselchen** [comment]
Hi there,
I just joined a new project "still" using go1.22.1 in their go.mod file. Locally however, I was already using 1.23.0.
golangci-lint itself was built on 1.22.1 as well. When running the linter on the project, I got "undefined" errors spanning the entire project, despite the project compiling just fine. downgrading go to 1.22.6 helped resolve the issue (I can't upgrade the project or recompile the linter)

If this incompatibility is intended, I would've greatly appreciated a message, instead of the incorrect error output.
Thanks!
